File: skiCentar/skiCentar.Model/TicketPurchase.cs
using System;

namespace skiCentar.Model
{
    public class TicketPurchase
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int TicketId { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public int Quantity { get; set; }

        public decimal TotalPrice { get; set; }

        public string? StripePaymentIntentId { get; set; }

        public int PaymentStatusId { get; set; }

        public virtual PaymentStatus PaymentStatus { get; set; } = null!;

        public virtual Ticket Ticket { get; set; } = null!;

        public virtual User User { get; set; } = null!;
    }
}

File: skiCentar/UI/skicentar_mobile/lib/models/ticket_purchase.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:skicentar_mobile/models/ticket.dart';

part 'ticket_purchase.g.dart';

@JsonSerializable()
class TicketPurchase{
  int? id;
  int? userId;
  int? ticketId;
  double? totalPrice;
  DateTime? purchaseDate;
  Ticket? ticket;

  TicketPurchase({this.id,this.userId,this.ticketId});

  factory TicketPurchase.fromJson(Map<String,dynamic> json) => _$TicketPurchaseFromJson(json);

  Map<String,dynamic> toJson() => _$TicketPurchaseToJson(this);
}